modules/nf-core/sickle/main.nf

```diff
@@ -0,0 +1,65 @@
+process SICKLE {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "bioconda::sickle-trim=1.33"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/sickle-trim:1.33--h7132678_7':
+        'quay.io/biocontainers/sickle-trim:1.33--h5bf99c6_6' }"
+
+    input:
+    tuple val(meta), path(reads)
+
+    output:
+    tuple val(meta), path("${prefix}.se.trimmed.fastq.gz"),        optional:true, emit: single_trimmed
+    tuple val(meta), path("${prefix}.pe{1,2}.trimmed.fastq.gz"),   optional:true, emit: paired_trimmed
+    tuple val(meta), path("${prefix}.singleton.trimmed.fastq.gz"), optional:true, emit: singleton_trimmed
+    tuple val(meta), path("*.log"), emit: log
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    prefix = task.ext.prefix ?: "${meta.id}"
+    def qual_type = "${meta.qual}"
```

Comment:
Is `qual_type` a mandatory input? Otherwise you should add this to `ext.args`

Reply:
It's also preffered to not force pipeline developers to use specific `meta` fields (`single_end` is the exception)

Reply:
> Is `qual_type` a mandatory input? Otherwise you should add this to `ext.args`

yes, "qual_type" is a mandatory argument (although by default it is set to "sanger" and it can take any of these three values: "solexa", "sanger" or "illumina". In the new PR, I removed "qual_type" from meta and put it as a separate input in input tuple. Do you have any suggestion on this? Should I completely remove "qual_type" ?

Reply:
> It's also preffered to not force pipeline developers to use specific `meta` fields (`single_end` is the exception)

I removed it from meta and put it as input in input tuple. Please review it. Thanks.

Reply:
I wouldn't add it as an input here if the tool automatically uses `sanger` as default and just let the user specify it in the `task.ext.args` then. The only reason `String` values should be given in the input is if the module 100% needs it to run (which isn't the case here)

Reply:
Sorry, my mistake! It is not default and the tool does not work without quality type, therefore, I suggest to keep it as one of the input in tuple. Apart from this, is there any other change that you recommend me to carry out?